lib/src/execution/windows/execution.cc

```diff
@@ -358,6 +358,19 @@ namespace facter { namespace execution {
         option_set<execution_options> const& options,
         uint32_t timeout)
     {
+        // Since we use a job object in the windows world, we want to
+        // be sure we're not in a job object, or at least able to
+        // break our processes out if we are in one.
+        BOOL in_job = FALSE;
+        IsProcessInJob(GetCurrentProcess(), nullptr, &in_job);
+        if (in_job) {
+            JOBOBJECT_BASIC_LIMIT_INFORMATION limits;
+            QueryInformationJobObject(nullptr, JobObjectBasicLimitInformation, &limits, sizeof(limits), nullptr);
+            if (!(limits.LimitFlags & JOB_OBJECT_LIMIT_BREAKAWAY_OK)) {
```

Comment:
We could, but that would break timeouts. I'd kind of rather fail hard instead of trying to stumble forward, especially since this should be pretty rare in the real world.